astropy/table/tests/test_mixin.py

```diff
@@ -756,56 +757,104 @@ def test_quantity_representation():
     ]
 
 
-def test_representation_representation():
+@pytest.mark.parametrize(
```

Comment:
note to reviewers: the diff is not minimal here because I started by refactoring as a parametrized test before I knew exactly how I was going to avoid duplication. In the end it's not necessary but I figure it doesn't hurt either, so I didn't undo it 🤷🏻‍♂️

Reply:
The refactored version is harder to read by human though. 🤔

Reply:
Well, happy to revert the non-essential part if requested.

Reply:
Let's see what @taldcroft says.

Reply:
The refactor is OK. It is slightly less readable but I think that is outweighed by the benefit of having the tests separated so if one fails the rest still run.